core/reader_re2.f

```diff
@@ -814,15 +816,25 @@ subroutine read_re2_hdr(ifbswap, ifverbose) ! open file & chk for byteswap
          call byte_read(hdr,20,ierr)
          if(ierr.ne.0) goto 100
 
-         read (hdr,1) version,nelgt,ldimr,nelgv
-    1    format(a5,i9,i3,i9)
+         read (hdr,'(a5)') version
+
+         if(version.eq.'#v004') then   
+            read (hdr,*) version,nelgt,ldimr,nelgv
+         else                          
+            read (hdr,1) version,nelgt,ldimr,nelgv
+         endif    
+   1     format(a5,i9,i3,i9)
+
+         if(ifverbose)
+     $      write (6,*) ' hdr calling ',version,nelgt,ldimr,nelgv
+             
  
          wdsizi = 4
          if(version.eq.'#v002') wdsizi = 8
-         if(version.eq.'#v003') then
-           wdsizi = 8
-           param(32)=1
-         endif
+         if(version.eq.'#v003') wdsizi = 8
+         if(version.eq.'#v004') wdsizi = 8
+
+         if(version.eq.'#v003') param(32)=1 !param(32)=0 read all BC from .re2
```

Comment:
Ethier we introduce with `#v004` the number of BC fields in the header and adjust param(32) or we assume that from `#v003` on there is only one boundary field.

Reply:
I simply kept the previous setting:  #v003 + param(32)=1, to avoid potential incompatibility.  
We can add nBCre2 if preferred. But this will involve more touches in re2 generating routines, needing another 20000-30000 node hours to test.

Reply:
`#v004` is changing the previous setting because it doesn't enforce `param(32) = 1` like `#v003`. This is confusing.